### PR TITLE
Fix build errors

### PR DESCRIPTION
**Pull request by lobkovilya**
Add required fields to Update struct

Signed-off-by: lobkovilya <lobkovilya@yandex.ru>

### CODE REVIEW

**Comment by mestery** [comment]
@lobkovilya I restarted the failing CircleCI build to see if it was transient or not (which it appeared to be).

**Comment by sbezverk** [comment]
@lobkovilya Quick question, what is the purpose of adding LocalMechanism to Update message?

**Comment by lobkovilya** [comment]
@sbezverk Actually I just see notice that LocalMechanism from Update structure already used in code, but structure doesn't contain it

**Comment by sbezverk** [comment]
@lobkovilya right, the reason we had RemoteMechanism in Update was to inform NSM about any possible changes to resources related to RemoteMechanisms. Having LocalMechanism in update means a sort of support for local configuration changes as well which is not implemented at all. So it will not be used for a while. If you really want to keep it, probably you need to add a TODO note, explaining that it is not used currently.

**Comment by sbezverk** [comment]
@lobkovilya not a problem, I am curious if you still have original build error, I would be interested to see what caused it.

File: dataplanes/vpp/pkg/nsmdataplane/nsmdataplane_server.go [hunk 1]
```diff
@@ -180,7 +181,7 @@ func (d DataplaneServer) MonitorMechanisms(empty *common.Empty, updateSrv datapl
 			d.remoteMechanisms = update.remoteMechanisms
 			d.localMechanisms = update.localMechanisms
 			if err := updateSrv.Send(&dataplaneapi.MechanismUpdate{
-				RemoteMechanisms: update.remoteMechanism,
+				RemoteMechanisms: update.remoteMechanisms,
```
Comment (edwarnicke) [inline dataplanes/vpp/pkg/nsmdataplane/nsmdataplane_server.go]: While you are at it, please also send

```LocalMechanisms: update.localMechanisms```

as well.

**Comment by sbezverk** [comment]
@lobkovilya scratch my comments, I just realized that @edwarnicke hacked out from the registration advertisement for available LocalMechanisms :(  big -1 about this.
So somebody hopefully @edwarnicke will need to implement new logic of NSM side.

**Comment by edwarnicke** [comment]
@sbezverk I believe we should be covered here: 
https://github.com/ligato/networkservicemesh/blob/master/plugins/dataplaneregistrar/dataplaneregistrar_reg.go#L89

on the NSM side.

Just like for the device plugin, register simply tells NSM how to contact the Dataplane, it calls MonitorMechanism on the Dataplane.

The Dataplane starts by [dumping its current Mechanism state](https://github.com/ligato/networkservicemesh/blob/master/dataplanes/vpp/pkg/nsmdataplane/nsmdataplane_server.go#L168), and then any updates.  Thus as soon as the NSM [calls dataplaneMonitor](https://github.com/ligato/networkservicemesh/blob/master/plugins/dataplaneregistrar/dataplaneregistrar_reg.go#L125), which [calls MonitorMechanism](https://github.com/ligato/networkservicemesh/blob/master/plugins/dataplaneregistrar/dataplaneregistrar_reg.go#L75) it updates with the [current state of the Dataplane Mechanisms](https://github.com/ligato/networkservicemesh/blob/master/plugins/dataplaneregistrar/dataplaneregistrar_reg.go#L89), and we keep in sync via streaming.

What I can't figure out is how the non-buildable piece being fixed here slipped past CI.  My guess is on my end it got mungled in the rebases... but CI should have broken...

**Comment by sbezverk** [comment]
@edwarnicke as I mentioned multiple times, imho removing capabilities advertisement in registration is not right. Example BGP, exchanges capabilities with its peers before establishing session, since you think differently I hope you will provide a alternative solution which will let NSM to learn about localMechanism capabilities.

**Comment by mestery** [comment]
Closing as we merged #353 instead.

**Pull request closed without merging**